ledger-service/http-json/src/main/scala/com/digitalasset/http/WebSocketService.scala

```diff
@@ -251,6 +279,40 @@ object WebSocketService {
           import spray.json._, JsonProtocol._
           "matchedQueries" -> p.toJson
         }
+
+      override def acsRequest(
+          maybePrefix: Option[domain.StartingOffset],
+          request: SearchForeverRequest,
+      ): Option[SearchForeverRequest] = {
+        import scalaz.std.list
+        val withoutOffset = request.queries.toList.filter(_.offset.isEmpty)
+        list.toNel(withoutOffset).map(SearchForeverRequest(_))
+      }
+
+      override def adjustRequest(
+          prefix: Option[domain.StartingOffset],
+          request: SearchForeverRequest,
+      ): SearchForeverRequest =
+        prefix.fold(request)(prefix =>
+          request.copy(
+            queries = request.queries.map(query =>
+              query.copy(offset = query.offset.orElse(Some(prefix.offset)))
+            )
+          )
+        )
+
+      import scalaz.syntax.foldable1._
+      import domain.Offset.orderingOptional
+
+      override def liveStartingOffset(
+          prefix: Option[domain.StartingOffset],
+          request: SearchForeverRequest,
+      ): Option[domain.StartingOffset] =
+        request.queries
```

Comment:
do you assume that the request has already been adjusted? probably worth documenting

Reply:
Addressed by https://github.com/digital-asset/daml/pull/9847/commits/893c5536383a64fda2d9ff54b0187f15af0c9707